### ISSUE TITLE
Revision timestamp

### ISSUE THREAD

**Issue by I-Am-Abacus** [collectiveidea/audited#363]
I am using this gem in a project using active record.

When I fetch the list of revisions of an object (#revisions method), the created_at and updated_at timestamps all show the values from the current record, not from the revision. This is obviously because created_at and updated_at are excluded from the list of attributes which are stored.

The audit object has a created_at attribute, IE the time of the revision. How do I access this from a returned revision instance?

Thanks, Martin

**Comment by Beartech** [comment]
Exactly!!! The thing that gets confusing is that the `.audits` method returns an array of single audit records (as it should), the `.revisions` returns an association that could be an array of arrays if more than one change was made in an audit record. I can see why but I don't get why this gem seems to be designed to NOT track when changes took place.

I ended up working around by creating a model method like this:

   ```
 def assignment_history      
      arr = self.audits.where("audited_changes ?| Array['user_id']").pluck(:created_at, "audited_changes -> 'user_id'")   #use the jsonb query to grab any audit records that record a :user_id change
      history = {}
      case 
      when arr.length == 1
          history[User.find(arr[0][1])] = self.created_at  #new items :user_id is an integer not an array i.e. [[Fri, 13 Apr 2018 21:42:48 UTC +00:00, 19]]
      when arr.length > 1
          history[User.find(arr[0][1][0])] = self.created_at  #grab the originally assigned user and the ppe created at date.
          arr.each do |r|
            history[User.find(r[1][1])] = r[0]    #add each subsequent user as a change with date
          end
    else
      history[self.user] = self.created_at
    end
    history
  end

```
This returns a Hash object that has the User object as a key and the date of assignment as the value. With that I can simply display it in HTML like:

```
<% @ppe.assignment_history.each do |user, date| %>
    Assigned to <%= user.last_first_name %> on <%= date.strftime("%F %T") %>
<% end %>
```

So I get something like:

Assigned to Moreing, Garry on   2018-03-27 23:39:33
Assigned to Kellog, Prent on       2018-03-27 23:39:51
Assigned to Juste, Geralda on    2018-03-28 00:32:53
Assigned to Overton, Briggs on  2018-03-28 00:33:25
Assigned to Mattedi, Jeffie on    2018-04-12 21:11:20
Assigned to Warehouse,  on       2018-04-14 00:02:39

**Comment by Beartech** [comment]
Ok, that code was awful and it was broken in certain examples, but mostly just due to having records in the DB that were created before I installed Audited. I knew there had to be an easy way to do this leveraging the built-n methods of Audited and here is it. This method goes in my model for the Equipment class. 

```
def assignment_history
  hh = self.audits.where("audited_changes ?| Array['user_id']").order(:version).pluck(:version)
  #this grabs the audit records that have user_id changes, saving their version numbers in an array
  hh.each do |v|
    history[self.revision(v).user] = self.audits.find_by(version: v).created_at
  end
  history
end
```
Now I can do 
`x = Equipment.find(1)`
`x.assignment_history`
=>
  {#<User:0x00007fcec1c61bb8> => Mon, 16 Oct 2017 04:03:36 UTC +00:00,   etc...}

Now I have my array without any fiddling with internal structures.
